libsolidity/analysis/TypeChecker.cpp

```diff
@@ -128,7 +126,7 @@ bool TypeChecker::visit(ContractDefinition const& _contract)
 	{
 		FixedHash<4> const& hash = it.first;
 		if (hashes.count(hash))
-			m_errorReporter.fatalTypeError(
+			m_errorReporter.typeError(
```

Comment:
I think this has to remain fatal, there could be some parts of the code that assume that the size of the hashes is the same as the number of functions.